sway-lsp/src/core/session.rs

```diff
@@ -63,19 +63,11 @@ impl Session {
         self.documents.contains_key(url.path())
     }
 
-    pub fn update_text_document(
-        &self,
-        url: &Url,
-        changes: Vec<TextDocumentContentChangeEvent>,
-    ) -> Result<(), DocumentError> {
-        match self.documents.get_mut(url.path()) {
-            Some(ref mut document) => {
-                changes.iter().for_each(|change| {
-                    document.apply_change(change);
-                });
-                Ok(())
-            }
-            _ => Err(DocumentError::DocumentNotFound),
+    pub fn update_text_document(&self, url: &Url, changes: Vec<TextDocumentContentChangeEvent>) {
+        if let Some(ref mut document) = self.documents.get_mut(url.path()) {
+            changes.iter().for_each(|change| {
+                document.apply_change(change);
+            });
```

Comment:
I guess once we get some logging we could do something like `log::warn!("no document for url: {}", url);` or something along those lines.

Reply:
Yeah exactly, logging warnings to the console makes more sense to report errors than crashing for an LSP server.